crates/rpc-server/src/facade.rs

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2024 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::fmt::Display;
+use std::sync::Arc;
+use std::time::Duration;
+
+use ccp_shared::nox_ccp_api::NoxCCPApi;
+use ccp_shared::proof::CCProof;
+use ccp_shared::proof::ProofIdx;
+use ccp_shared::types::CUAllocation;
+use ccp_shared::types::EpochParameters;
+
+use eyre::Context;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::TryRecvError;
+use tokio::sync::Mutex;
+use tokio::time::timeout;
+
+const WAIT_LOCK_DURATION: Duration = Duration::from_secs(2);
```

Comment:
Too many, it takes 2 secs to initialize dataset only on ARM MacBook, on server linux x86 systems  it's less than 1 sec, so let's put 1 sec here.

Reply:
Ah, I got the semantics, I don't think that we need `timeout` for getting proofs, let's better do try_lock and return a busy status for Nox, because otherwise it'd be not so sustainable.

Reply:
I think it is needed so that concurrent get_proofs_after would work.

Reply:
However, it may be resolved with a RwLock.